server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -554,6 +554,10 @@ export class AgenticChatController implements ChatHandlers {
                         break
                 }
             } catch (err) {
+                if (CancellationError.isUserCancelled(err)) {
```

Comment:
I handle tool validation rejections very similarly [here](https://github.com/aws/language-servers/pull/1056). Do you think their logic should be combined more? 

I think they should be separate since the stop button and the reject button are different, and could warrant different behavior in the future, but open to changing it.

Reply:
I think we should separate them since their intentions are different. I think they'd just have a minor overlap